### ISSUE TITLE
Discuss whether AWS SDK needs to be bundled

### ISSUE THREAD

**Issue by skaffille** [mkimuram/k8sviz#36]
In the line with #29 

```
RUN apk add --no-cache \
        python3 \
        py3-pip \
    && pip3 install --upgrade pip \
    && pip3 install \
        awscli \
    && rm -rf /var/cache/apk/*
```

**Comment by mkimuram** [comment]
@skaffille 

Thank you for opening this issue.

Could you explain why AWS SDK will be needed, like how it is used in k8sviz for which purpose?

**Comment by 2tim** [comment]
You could work around this by creating a kube service account and generating a kubeconfig file that has the embedded token for authn in the file then passing it to the script as your kubeconfig file. I'm planning on trying this now.

**Comment by mkimuram** [comment]
@2tim 

Thank you for your explanation. I didn't know this use case and very good to know it.
I understand the documents for them are as follows:

> I bumped into this myself using the AWS auth method for EKS. Its a specific use case where proxies the user identity call through the aws cli.

https://docs.aws.amazon.com/eks/latest/userguide/managing-auth.html

> You could work around this by creating a kube service account and generating a kubeconfig file that has the embedded token for authn in the file then passing it to the script as your kubeconfig file.

https://docs.aws.amazon.com/eks/latest/userguide/create-kubeconfig.html

> I'm planning on trying this now.

Thank you very much for working on it. I will wait for your report.
Also, I will consider whether handling above use case directly will be good to go (I still need to understand the detail and need to check whether this kind of commands are allowed to be included in the container images and to be distributed publicly).

**Comment by mkimuram** [comment]
@skaffille @2tim 

Created #44 to provide a way to include AWS SDK.
Could you give feedback if it would be of help (and test if the image built with the command works well, if possible)?

**Comment by mkimuram** [comment]
Merged #44. I hope that below works for environments that requires AWS SDK:
```bash
$ DEVEL_IMAGE=mkimuram/k8sviz DEVEL_TAG=$(cat version.txt) TARGET=aws make image-build
```
And below works for Google cloud SDK:
```bash
 $ DEVEL_IMAGE=mkimuram/k8sviz DEVEL_TAG=$(cat version.txt) TARGET=gcloud make image-build
```

**Comment by Sreddy-gs** [comment]
@mkimuram i am going to try that and let you know thank you

**Comment by Sreddy-gs** [comment]
@mkimuram Getting this error "Unable to locate credentials. You can configure credentials by running "aws configure".
Failed to get namespace "default": Get "https://7*/default": getting credentials: exec: executable aws failed with exit code 255", though i have passed aws credentials to the docker image

May be need to have more parameters in shell script to add aws credentials

**Comment by mkimuram** [comment]
@Sreddy-gs Yes, Maybe something like below will be needed around [here](https://github.com/mkimuram/k8sviz/blob/master/k8sviz.sh#L72).

```
-v ~/.aws/config:/awsconfig:ro
-e AWS_CONFIG_FILE=/awsconfig
```

Could you try?

**Comment by Sreddy-gs** [comment]
@mkimuram that worked :) thank you
